### PR TITLE
Minor Registry UX enhancements

### PR DESCRIPTION
**Pull request by alexyaang**
Closes https://github.com/microsoft/vscode-docker/issues/4082
Closes https://github.com/microsoft/vscode-docker/issues/4083
Closes https://github.com/microsoft/vscode-docker/issues/4087

### CODE REVIEW

File: src/tree/registries/UnifiedRegistryTreeDataProvider.ts [hunk 1]
```diff
@@ -74,17 +70,17 @@ export class UnifiedRegistryTreeDataProvider implements vscode.TreeDataProvider<
...
                         wrappedItem: r,
                         parent: undefined
                     };
                 }));
             }
-
-            return unifiedRoots;
         }
+
+        return unifiedRegistryItems.sort((a, b) => (a.wrappedItem as CommonRegistryItem).label.localeCompare((b.wrappedItem as CommonRegistryItem).label));
```
Comment (bwateratmsft) [inline src/tree/registries/UnifiedRegistryTreeDataProvider.ts:83]: I think we need a more advanced locale compare function. Even though all of our current registry providers use `CommonRegistryItem`, it's not a strict requirement. So what I'd do is try getting their labels if they are `CommonRegistryItems`, but if not, sort by their `toString` results.

Comment (alexyaang) [reply src/tree/registries/UnifiedRegistryTreeDataProvider.ts:83]: yeah i like that!

Comment (alexyaang) [reply src/tree/registries/UnifiedRegistryTreeDataProvider.ts:83]: should we add a `isCommonRegistryItem()` in the package then?

Comment (alexyaang) [reply src/tree/registries/UnifiedRegistryTreeDataProvider.ts:83]: or can we do `instanceof`

Comment (bwateratmsft) [reply src/tree/registries/UnifiedRegistryTreeDataProvider.ts:83]: I'm surprised we haven't already got an `isCommonRegistryItem` method but that would be the way to go.

**Review by bwateratmsft — APPROVED** [review]

**Pull request merged**